### PR TITLE
Add go tool cgo wrapper

### PR DESCRIPTION
**Pull request by ianthehat**
Does all the work of filtering and choosing cgo sources, replaces all the ugly shell script.
This is a prerequisite for the cgo_library merge.

### CODE REVIEW

File: go/tools/builders/cgo.go [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+	gotool := args[0]
+	if err := flags.Parse(args[1:]); err != nil {
+		return err
+	}
+
+	// apply build constraints to the source list
+	// also pick out the cgo sources
+	bctx := build.Default
+	bctx.CgoEnabled = true
+	cgo_srcs := []string{}
```
Comment (jayconrod) [inline go/tools/builders/cgo.go]: Nit: camelCase

File: go/tools/builders/cgo.go [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+		if !strings.HasSuffix(in, ".go") {
+			// Not a go file, just filter, assume C or C-like
+			if match {
+				// not filtered, copy over
+				if err := ioutil.WriteFile(out, data, 0644); err != nil {
+					return err
+				}
+			} else {
+				// filtered, make empty file
+				if err := ioutil.WriteFile(out, []byte("//Empty file"), 0644); err != nil {
```
Comment (jayconrod) [inline go/tools/builders/cgo.go]: I would actually leave these files empty. // may not be a valid comment prefix for assembly dialects compiled by the external compiler.

File: go/tools/builders/cgo.go [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+					return err
+				}
+			}
+			continue
+		}
+
+		// Go source, must produce both c and go outputs
+		cOut := strings.TrimSuffix(out, ".cgo1.go") + ".cgo2.c"
+		if err != nil {
```
Comment (jayconrod) [inline go/tools/builders/cgo.go]: Error check seems out of place. Did you mean to check that the suffix was trimmed?

File: go/tools/builders/cgo.go (hunk 1 above)
Comment (ianthehat) [inline go/tools/builders/cgo.go]: Done, bouncing between skylark and go I think I am going to make a lot of these mistakes :)

File: go/tools/builders/cgo.go (hunk 2 above)
Comment (ianthehat) [inline go/tools/builders/cgo.go]: Done. The go toolchain presumes all assembler files support c style comments for build tags, but it's probably safer anyway.

File: go/tools/builders/cgo.go (hunk 3 above)
Comment (ianthehat) [inline go/tools/builders/cgo.go]: Nope, just left over from when there was another bit of code in there, removed.

**Pull request merged**